### PR TITLE
host-ctr: all devices nodes allowed when superpowered

### PR DESCRIPTION
**Pull request by etungsten**
**Issue number:**
N/A

**Description of changes:**
```
Author: Erikson Tung <etung@amazon.com>
Date:   Thu Aug 13 09:44:14 2020 -0700

    host-ctr: all devices nodes allowed when superpowered
    
    Allow access to all devices nodes when a container is superpowered

```

**Testing done:**
No longer getting permission errors when trying to access the partition.
```
[ec2-user@ip-192-168-19-171 ~]$ sudo sheltie
bash-5.0# signpost status
OS disk: /dev/nvme0n1
Set A:   boot=/dev/nvme0n1p2 root=/dev/nvme0n1p3 hash=/dev/nvme0n1p4 priority=1 tries_left=0 successful=true
Set B:   boot=/dev/nvme0n1p6 root=/dev/nvme0n1p7 hash=/dev/nvme0n1p8 priority=0 tries_left=0 successful=false
Active:  Set A
Next:    Set A
bash-5.0# dd if=/dev/nvme0n1 of=/tmp/foo bs=4k count=8 
8+0 records in
8+0 records out
32768 bytes (33 kB, 32 KiB) copied, 6.5269e-05 s, 502 MB/s
bash-5.0# updog update -i 0.4.1 -r -n                  
Starting update to 0.4.1
Cannot schedule shutdown without logind support, proceeding with immediate shutdown.
```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Comment by etungsten** [comment]
Push above is for changes after `go mod tidy`

File: sources/host-ctr/cmd/host-ctr/main.go [hunk 1]
```diff
@@ -371,6 +371,26 @@ func withMountLabel(label string) oci.SpecOpts {
 	}
 }
 
+// withAllDevicesAllowed permits all access on all devices nodes for the container
+// Taken from https://github.com/containerd/containerd/blob/master/oci/spec_opts.go#L1020
+// which is not availabe in containerd v1.3.7
```
Comment (bcressey) [inline sources/host-ctr/cmd/host-ctr/main.go]: nit: typo, "available"

File: sources/host-ctr/cmd/host-ctr/main.go [hunk 2]
```diff
@@ -371,6 +371,26 @@ func withMountLabel(label string) oci.SpecOpts {
 	}
 }
 
+// withAllDevicesAllowed permits all access on all devices nodes for the container
+// Taken from https://github.com/containerd/containerd/blob/master/oci/spec_opts.go#L1020
```
Comment (jahkeup) [inline sources/host-ctr/cmd/host-ctr/main.go]: Please update this with a commit perma-link for whence this was taken.

**Comment by etungsten** [comment]
Push above addresses @bcressey and @jahkeup 's comments.

**Comment by etungsten** [comment]
Separated out `withAllDevicesAllowed` to its file with a `SPDX-License-Identifier` at the top for Apache-2.0

File: sources/host-ctr/cmd/host-ctr/all_devices_allowed.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+	"github.com/containerd/containerd/containers"
+	"github.com/containerd/containerd/oci"
+	runtimespec "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+// withAllDevicesAllowed permits all access on all devices nodes for the container
+// Taken from https://github.com/containerd/containerd/blob/25947db049b058fcbce291ef883b8b512e3ea440/oci/spec_opts.go#L1010
+// which is not available in containerd v1.3.7
+// TODO Once we move to containerd v1.4.0 remove this and use oci.WithAllDevicesAllowed
```
Comment (iliana) [inline sources/host-ctr/cmd/host-ctr/all_devices_allowed.go]: Can this become an issue, and we can reference the issue ID here?

**Review by samuelkarp — APPROVED** [review]

**Review by iliana — APPROVED** [review]
OK with regards to attribution on `all_devices_allowed.go`

File: sources/host-ctr/cmd/host-ctr/all_devices_allowed.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"context"
+
+	"github.com/containerd/containerd/containers"
+	"github.com/containerd/containerd/oci"
+	runtimespec "github.com/opencontainers/runtime-spec/specs-go"
```
Comment (bcressey) [inline sources/host-ctr/cmd/host-ctr/all_devices_allowed.go]: nit: if we import this as "specs" then it will get rid of most of the delta from the source

**Review by bcressey — APPROVED** [review]

File: sources/host-ctr/cmd/host-ctr/all_devices_allowed.go (hunk 3 above)
Comment (etungsten) [reply sources/host-ctr/cmd/host-ctr/all_devices_allowed.go]: https://github.com/bottlerocket-os/bottlerocket/issues/1038

**Comment by etungsten** [comment]
Push above adds a comment describing the modification to `withAllDevicesAllowed` and changes `runtimespec` to `spec` to better align with original source.

**Review by jahkeup — APPROVED** [review]
If we're sticking with the drop-in function route, then.. ship it! :+1:

**Review by bcressey — APPROVED** [review]

**Pull request merged**